invokeai/app/services/invocation_queue.py

```diff
@@ -1,31 +1,17 @@
 # Copyright (c) 2022 Kyle Schouviller (https://github.com/kyle0654)
 
+import time
 from abc import ABC, abstractmethod
+from dataclasses import dataclass
 from queue import Queue
-import time
 
 
-# TODO: make this serializable
+@dataclass
 class InvocationQueueItem:
-    # session_id: str
     graph_execution_state_id: str
     invocation_id: str
     invoke_all: bool
-    timestamp: float
-
-    def __init__(
-        self,
-        # session_id: str,
-        graph_execution_state_id: str,
-        invocation_id: str,
-        invoke_all: bool = False,
-    ):
-        # self.session_id = session_id
-        self.graph_execution_state_id = graph_execution_state_id
-        self.invocation_id = invocation_id
-        self.invoke_all = invoke_all
-        self.timestamp = time.time()
-
+    timestamp: float = time.time()
```

Comment:
I don't think this will work correctly. If you use Pydantic, you'd set `time.time` as the `default_factory` so it would get the current time at construction.

Reply:
thank you - let me look into that. i think because the `@dataclass` decorator creates the `__init__` method, those will actually be instance variables (but i'm not sure and will test). regardless, i'll make it a Pydantic class as you suggest. Anything else I need to do to make it serializable for saving to Redis if i go that route? Was there a reason it wasn't already a Pydantic class?

Reply:
No reason, just didn't need to at the time.

Reply:
Good catch re: timestamp - you're right, it was initialized as a class variable.

Converted to Pydantic class and it's working well